tests/test_fork_provider.py

```diff
@@ -80,10 +84,8 @@ def test_reset_fork(networks, mainnet_fork_provider):
     assert block_num_after_reset < prev_block_num
 
 
-def test_transaction_contract_as_sender(network_api, contract_instance, provider):
-    provider = get_hardhat_provider(network_api)
-    provider.connect()
-    networks.active_provider = provider
-    assert contract_instance
-    # contract_instance.set_number(10, sender=contract_instance)
-    # assert contract_instance.my_number() == 10
+@alchemy_xfail
+def test_transaction_contract_as_sender(networks, network_api, contract_instance):
+    with pytest.raises(ContractLogicError):
```

Comment:
Oh also, because of the logic in the contract, that always reverts. but that is ok. Task failed successfully.